### PR TITLE
Align friendly names uniqueization

### PR DESCRIPTION
**Pull request by itikhono**
### Details:
Removed code that makes friendly names unique from Serialization and a name uniqueness check from Deserializator.
Enabled the mode of ResolveNameCollisions transformation to uniqueize all friendly names, not only autogenerated in Frontends

### Tickets:
 - *CVS-131567*

### CODE REVIEW

**Review by rkazants — COMMENTED** [review]
When can we have a case that friendly names are not unique?
Friendly name are mostly about original framework name, can you share a (TF, PT) model with such case?

**Comment by itikhono** [comment]
> When can we have a case that friendly names are not unique? Friendly name are mostly about original framework name, can you share a (TF, PT) model with such case?

I tried to delete the code that makes friendly names unique from Serialization and leave the exception in Deserializer. But some e2e tests failed:
https://github.com/openvinotoolkit/openvino/pull/22618

It have to be investigated. But this case proves that we can't guarantee name uniqueness without enabling ResolveNameCollisions to fix all friendly names. And even this is not 100% guarantee, but it makes situation more stable and predictable.

**Comment by itikhono** [comment]
We decided to run calibration/perf validation for this PR before merging, waiting for results

File: src/frontends/ir/src/frontend.cpp [hunk 1]
```diff
@@ -249,7 +249,7 @@ std::string FrontEnd::get_name() const {
 
 void FrontEnd::normalize(const std::shared_ptr<ov::Model>& model) const {
     ov::pass::Manager manager;
-    manager.register_pass<pass::ResolveNameCollisions>();
+    manager.register_pass<pass::ResolveNameCollisions>(true);
```
Comment (slyalin) [inline src/frontends/ir/src/frontend.cpp:252]: Why? Will it make names from IR unique? It is not what we discussed.

Comment (itikhono) [reply src/frontends/ir/src/frontend.cpp]: done

File: src/frontends/tensorflow/src/frontend.cpp [hunk 2]
```diff
@@ -550,7 +550,7 @@ void FrontEnd::normalize(const std::shared_ptr<ov::Model>& model) const {
     manager.register_pass<ov::pass::RemoveConcatZeroDimInput>();
     manager.register_pass<ov::pass::TransposeSinkingGeneral>();
     manager.register_pass<ov::pass::ReverseShapeAndTypeInfer>();
-    manager.register_pass<ov::pass::ResolveNameCollisions>();
+    manager.register_pass<ov::pass::ResolveNameCollisions>(true);
```
Comment (rkazants) [inline src/frontends/tensorflow/src/frontend.cpp:553]: please clarify,  why do we need to resolve uniqueness for non-autogenerated names. TF should provide unique names for nodes

Comment (itikhono) [reply src/frontends/tensorflow/src/frontend.cpp:553]: TF Frontend can create additional friendly names during conversion

Comment (rkazants) [reply src/frontends/tensorflow/src/frontend.cpp:553]: if it creates additional friendly names, it means they are auto-generated. For auto-generated names, this transformation already works. Please correct me if I am wrong.

Comment (itikhono) [reply src/frontends/tensorflow/src/frontend.cpp:553]: There's no guarantee this won't create duplicates if someone adds code manually setting friendly names.

Comment (itikhono) [reply src/frontends/tensorflow/src/frontend.cpp:553]: And maybe somehow it's possible to create duplicates in TF, some workaround or trick

**Review by rkazants — APPROVED** [review]

File: src/frontends/tensorflow/src/frontend.cpp (hunk 2 above)
Comment (praasz) [reply src/frontends/tensorflow/src/frontend.cpp:553]: Can some test be added to simulate the "trick" that name will be duplicated?

**Review by praasz — APPROVED** [review]

**Comment by slyalin** [comment]
> We decided to run **calibration**/perf validation for this PR before merging, waiting for results

Does it mean, that it was verified with NNCF flow for regressions? Everything was passed?

**Comment by itikhono** [comment]
> > We decided to run **calibration**/perf validation for this PR before merging, waiting for results
> 
> Does it mean, that it was verified with NNCF flow for regressions? Everything was passed?

I will double check, it looks like, the results are still available

**Pull request merged**